lib/src/binary_codec.dart

```diff
@@ -475,6 +568,18 @@ class PostgresBinaryDecoder extends Converter<Uint8List, dynamic> {
       }
       weight--;
     }
-    return '$sign${intPart.replaceAll(RegExp(r'^0+'), '')}.${fractPart.padRight(dScale, '0').substring(0, dScale)}';
+
+    if (weight >= 0) {
+      // Trailing zeros were omitted
+      intPart += '0000' * (weight + 1);
+    }
+
+    var result = '$sign${intPart.replaceAll(RegExp(r'^0+'), '')}';
```

Comment:
Let's move the `RegExp` to a top-level final field, e.g. `_leadingZerosRegExp`. That way it will be initialized and parsed only once.